dashboard/src/modules/components/TableOfContent/index.jsx

```diff
@@ -209,8 +185,55 @@ const TableOfContent = () => {
       </Breadcrumb>
     );
   };
+  const getMyBreadCrumbClick = () => {
+    drillOut("rootMenu", "group:start_rollout", null);
+    setStack(1);
+    setTableData(stack[0].files);
+    setSearchSpace(stack[0].files);
+    setBreadCrumb(initialBreadcrumb([]));
+    setParam("");
+    setBreadCrumbLabels([]);
+  };
   const getSubFolderData = (data) => {
-    dispatch(fetchTOC(params["*"], `/${data}`, true));
+    dispatch(fetchTOC(params["dataset_id"], `/${data}`, true));
+  };
+  const attachBreadCrumbs = (data, firstHierarchyLevel) => {
+    breadCrumbLabels.push(data);
+    setBreadCrumbLabels(breadCrumbLabels);
+    setBreadCrumb(
+      firstHierarchyLevel
+        ? initialBreadcrumb(breadCrumbLabels)
+        : appGroupingBreadcrumb(false, breadCrumbLabels)
+    );
+    const dirPath = param.concat(`${firstHierarchyLevel ? "" : "/"}`, data);
+    setParam(dirPath);
+    setIsLoading(true);
+    getSubFolderData(dirPath);
+  };
+  const updateHighlightedRow = (index) => {
+    if (index >= page * perPage) {
+      setPage(
+        page +
+          Math.ceil(
+            (index -
+              (activeFile === -1
+                ? perPage - 1
+                : Math.floor(activeFile / perPage) * perPage + perPage - 1)) /
+              perPage
+          )
+      );
+      setActiveFile(index);
+    } else if (index < (page - 1) * perPage) {
+      setPage(
+        page -
+          Math.ceil(
+            (Math.floor(activeFile / perPage) * perPage - index) / perPage
+          )
+      );
+      setActiveFile(index);
+    } else {
+      setActiveFile(index);
+    }
   };
```

Comment:
If I might, would this work?
```
  const updateHighlightedRow = (index) => {
    const newPage = Math.floor(index / perPage);
    if (newPage != page) {
      setPage(newPage);
    }
    setActiveFile(index);
  };
```
I think it would be better....